File: backend/internal/models/trip.go
package models

import (
    "time"
)

type Trip struct {
    ID                int       `json:"id" db:"id"`
    DriverID          int       `json:"driverId" db:"driver_id"`
    FromLocation      string    `json:"from" db:"from_location"`
    ToLocation        string    `json:"to" db:"to_location"`
    DepartureTime     time.Time `json:"departureTime" db:"departure_time"`
    MaxPassengers     int       `json:"maxPassengers" db:"max_passengers"`
    CurrentPassengers int       `json:"currentPassengers" db:"current_passengers"`
    PricePerPerson    float64   `json:"pricePerPerson" db:"price_per_person"`
    Description       *string   `json:"description,omitempty" db:"description"`
    Status            string    `json:"status" db:"status"`
    CreatedAt         time.Time `json:"createdAt" db:"created_at"`
    UpdatedAt         time.Time `json:"updatedAt" db:"updated_at"`
    
    // Relationships
    Driver     *User  `json:"driver,omitempty"`
    Passengers []User `json:"passengers,omitempty"`
    UserRole  string `json:"userRole,omitempty"`
}

type TripSearchCriteria struct {
    From          string  `json:"from"`
    To            string  `json:"to"`
    DepartureDate string  `json:"departureDate"`
    MaxPrice      float64 `json:"maxPrice"`
    Limit         int     `json:"limit"`
    Offset        int     `json:"offset"`
}

File: frontend/src/types/trip.ts
import { User } from './user'

export interface Trip {
  id: number
  driverId: number
  from: string
  to: string
  departureTime: string
  maxPassengers: number
  currentPassengers: number
  pricePerPerson: number
  description?: string
  status: 'active' | 'completed' | 'cancelled'
  createdAt: string
  updatedAt: string
  driver?: User
  passengers?: User[]
  userRole?: 'driver' | 'passenger' | 'none' 
}

export interface CreateTripData {
  from: string
  to: string
  departureTime: string
  maxPassengers: number
  pricePerPerson: number
  description?: string
}

export interface UpdateTripData {
  from?: string
  to?: string
  departureTime?: string
  maxPassengers?: number
  pricePerPerson?: number
  description?: string
}

export interface SearchTripsData {
  from?: string
  to?: string
  departureDate?: string
  maxPrice?: number
  limit?: number
  offset?: number
}

export interface TripMatch {
  id: number
  tripId: number
  passengerId: number
  status: 'pending' | 'confirmed' | 'cancelled'
  joinedAt: string
  trip?: Trip
  passenger?: User
}

export interface TripStats {
  totalTrips: number
  activeTrips: number
  completedTrips: number
  cancelledTrips: number
  totalEarnings: number
  averageRating: number
}

export interface TripFilters {
  status?: string[]
  dateRange?: {
    start: string
    end: string
  }
  priceRange?: {
    min: number
    max: number
  }
  locations?: string[]
}

export interface TripSortOptions {
  field: 'departureTime' | 'pricePerPerson' | 'createdAt'
  direction: 'asc' | 'desc'
}

export interface PaginatedTripsResponse {
  trips: Trip[]
  total: number
  page: number
  limit: number
  hasNext: boolean
  hasPrev: boolean
}
